### PR TITLE
features/transcoding update

### PR DESCRIPTION
**Pull request by sentriz**
**update**
added ui for select profiles on a per user/client basis

in future it would be nice for the "client name" field to be a dropdown from previously seen clients. and also maybe the transcode profiles be configurable from an admin ui. but this is a good start i think.

also, there are some very unrelated changes in this pr - sorry about that 

if you're happy to merge this, i'm happy merge the main pr. but ofc let me know if you're okay with what's here

### CODE REVIEW

**Comment by spijet** [comment]
Do I understand it right that if user specifies an unknown client name (not defined in this list), or doesn't send a `c=` parameter at all, Gonic will respond with a "raw" untranscoded track, instead of falling back to some default transcoding profile?

Also, a good idea for the next step: add a similar table for transcoding profiles :D

ADD — seems like it *does* fall back to raw stream:
```golang
if gorm.IsRecordNotFoundError(err) {
	serveTrackRaw(w, r, servOpts)
	return nil
}
```

I think this is a bad idea, especially for mobile clients who may be connecting over cellular network. :)
Maybe we could add `raw` as a special-case encoding "profile" (that actually would redirect to `serveTrackRaw()`), so the user could explicitly enable it if they wanted to?

**Comment by sentriz** [comment]
Yeah that's right. IMO streaming raw should be the default. I like to stream flac sometimes, and some devices like raspberry pi mightn't be too happy about transcoding

**Comment by spijet** [comment]
That's a good point. The VM I use at home is pretty low-spec too. :P
Actually, that's why I came up with transcoding cache idea in the first place, back when I tried to implement it as an "encoder" executable for Java-based Subsonic versions. :)

Maybe it's possible to add a switch that controls the default behaviour ("stream raw" or "stream using fallback transcoding profile")?

**Comment by sentriz** [comment]
how about this functionality:
- user has no defined rules: stream raw
- user has two normal rules defined (say for dsub and soundwaves): dsub and soundwaves transcode with selected profile, raw everything else
- same as above but added user has defined an extra "*" client: dsub and soundwaves transcode with selected profile, transcode everyting else

**Comment by spijet** [comment]
Sounds like a plan!

**Comment by sentriz** [comment]
done :)

File: server/ctrlsubsonic/handlers_raw.go [hunk 1]
```diff
@@ -118,14 +118,16 @@ func (c *Controller) ServeStream(w http.ResponseWriter, r *http.Request) *spec.R
 		play.Count++           // for getAlbumList?type=frequent
 		c.DB.Save(&play)
 	}()
-	client := params.GetOr("c", "*")
+	client := params.Get("c")
```
Comment (sentriz) [inline server/ctrlsubsonic/handlers_raw.go:121]: i remembered that the `c` param is mandatory anyway, so this will never be `""`

Comment (spijet) [reply server/ctrlsubsonic/handlers_raw.go:121]: :+1:

**Comment by spijet** [comment]
LGTM, let's merge :)

**Pull request merged**